lib/rules/no-new-func.js

```diff
@@ -29,26 +29,25 @@ module.exports = {
 
     create(context) {
 
-        //--------------------------------------------------------------------------
-        // Helpers
-        //--------------------------------------------------------------------------
-
-        /**
-         * Reports a node.
-         * @param {ASTNode} node The node to report
-         * @returns {void}
-         * @private
-         */
-        function report(node) {
-            context.report({
-                node,
-                messageId: "noFunctionConstructor"
-            });
-        }
-
         return {
-            "NewExpression[callee.name = 'Function']": report,
-            "CallExpression[callee.name = 'Function']": report
+            "Program:exit"() {
+                const globalScope = context.getScope();
+                const variable = globalScope.set.get("Function");
+
+                if (variable && variable.defs.length === 0) {
+                    variable.references.forEach(ref => {
+                        const node = ref.identifier;
+                        const { parent } = node;
+
+                        if (parent && (parent.type === "NewExpression" || parent.type === "CallExpression")) {
```

Comment:
We should check whether it is `callee`. Without that, this would a false positive:

```js
/* eslint no-new-func: ["error"] */

foo(Function);
```